pkg/certificates/renewerubuntu.go

```diff
@@ -0,0 +1,117 @@
+package certificates
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/aws/eks-anywhere/pkg/logger"
+)
+
+// LinuxRenewer implements OSRenewer for Linux-based systems (Ubuntu and RHEL).
+type LinuxRenewer struct {
+	certPaths CertificatePaths
+	osType    OSType
+}
+
+// NewLinuxRenewer creates a new LinuxRenewer with the specified certificate paths and OS type.
+func NewLinuxRenewer(paths CertificatePaths, osType OSType) *LinuxRenewer {
+	return &LinuxRenewer{certPaths: paths, osType: osType}
+}
+
+// RenewControlPlaneCerts renews control plane certificates on a Linux node.
+func (l *LinuxRenewer) RenewControlPlaneCerts(ctx context.Context, node string, cfg *RenewalConfig, component string, ssh SSHRunner, backupDir string) error {
+	logger.V(2).Info(fmt.Sprintf("Processing node %s...", node))
+
+	hasExternalEtcd := cfg != nil && len(cfg.Etcd.Nodes) > 0
+
+	builder := NewLinuxControlPlaneCommandBuilder(l.certPaths, backupDir, component, hasExternalEtcd)
```

Comment:
For now, can we have Backup, Renew, Validate and Restart inside this file and remove entire dependency on NewLinuxControlPlaneCommandBuilder.